src/Helper/DownloadLinkUriGenerator.php

```diff
@@ -35,6 +35,6 @@ public function check(string $uri) : DownloadLink
         $name = array_pop($path);
         $uri = array_pop($path);
 
-        return new DownloadLink(base64_decode($uri), $name);
+        return new DownloadLink(base64_decode(strtr($uri, '._-', '+/=')), $name);
```

Comment:
would extract the two strings into two constants at the top of this class, since they are duplicated across `generate()` and `check()` and must match

Reply:
Using `urlencode()` instead. This should resolve this issue.